sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakeFileAsyncClient.java

```diff
@@ -0,0 +1,437 @@
+     *
+     * <p><strong>Code Samples>Code Samples</strong></p>
+     *
+     * {@codesnippet com.azure.storage.file.datalake.DataLakeFileAsyncClient.append#Flux-long-long}
+     *
+     * <p>For more information, see the
+     * <a href="https://docs.microsoft.com/en-us/rest/api/storageservices/datalakestoragegen2/path/update">Azure
+     * Docs</a></p>
+     *
+     * @param data The data to write to the file.
+     * @param offset The position where the data is to be appended.
+     * @param length The exact length of the data. It is important that this value match precisely the length of the
+     * data emitted by the {@code Flux}.
+     *
+     * @return A reactive response signalling completion.
+     */
+    public Mono<Void> append(Flux<ByteBuffer> data, long offset, long length) {
+        try {
+            return appendWithResponse(data, offset, length, null, null).flatMap(FluxUtil::toMono);
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    /**
+     * Appends data to the specified resource to later be flushed (written) by a call to flush
+     *
+     * <p><strong>Code Samples>Code Samples</strong></p>
+     *
+     * {@codesnippet com.azure.storage.file.datalake.DataLakeFileAsyncClient.appendWithResponse#Flux-long-long-byte-String}
+     *
+     * <p>For more information, see the
+     * <a href="https://docs.microsoft.com/en-us/rest/api/storageservices/datalakestoragegen2/path/update">Azure
+     * Docs</a></p>
+     *
+     * @param data The data to write to the file.
+     * @param offset The position where the data is to be appended.
+     * @param length The exact length of the data. It is important that this value match precisely the length of the
+     * data emitted by the {@code Flux}.
+     * @param contentMd5 An MD5 hash of the content of the data. If specified, the service will calculate the MD5 of the
+     * received data and fail the request if it does not match the provided MD5.
+     * @param leaseId By setting lease access conditions, requests will fail if the provided lease does
+     * not match the active lease on the file.
+     *
+     * @return A reactive response signalling completion.
+     */
+    public Mono<Response<Void>> appendWithResponse(Flux<ByteBuffer> data, long offset, long length,
+        byte[] contentMd5, String leaseId) {
+        try {
+            return withContext(context -> appendWithResponse(data, offset, length, contentMd5,
+                leaseId, context));
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    Mono<Response<Void>> appendWithResponse(Flux<ByteBuffer> data, long offset, long length,
+        byte[] contentMd5, String leaseId, Context context) {
+
+        LeaseAccessConditions leaseAccessConditions = new LeaseAccessConditions().setLeaseId(leaseId);
+
+        PathHttpHeaders headers = new PathHttpHeaders().setContentMd5(contentMd5);
+
```

Comment:
This is strange to me. I haven't looked at the test, so maybe this is working as expected, but the value in PathHttpHeaders should translate to the x-ms-content-md5 that is set on the blob. There should be a separate parameter that is the transactionalMd5 corresponding to the Content-MD5 header

Reply:
You're right. There's some transactional hash method in PathHttpHeaders, I'll use that instead.